libraries/AP_Airspeed/AP_Airspeed_SDP3X.cpp

```diff
@@ -0,0 +1,207 @@
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+  backend driver for airspeed from a I2C SDP3X sensor
+ */
+#include "AP_Airspeed_SDP3X.h"
+
+#include <AP_Common/AP_Common.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/I2CDevice.h>
+#include <AP_Math/AP_Math.h>
+#include <stdio.h>
+#include <utility>
+
+extern const AP_HAL::HAL &hal;
+
+AP_Airspeed_SDP3X::AP_Airspeed_SDP3X(AP_Airspeed &_frontend) :
+    AP_Airspeed_Backend(_frontend)
+{
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_SDP3X::init()
+{
+    const struct{
+        uint8_t bus;
+        uint8_t addr;
+    } addresses[] = {
+        { 1, SDP3XD0_I2C_ADDR },
+        { 0, SDP3XD0_I2C_ADDR },
+        { 2, SDP3XD0_I2C_ADDR },
+        { 1, SDP3XD1_I2C_ADDR },
+        { 0, SDP3XD1_I2C_ADDR },
+        { 2, SDP3XD1_I2C_ADDR },
+        { 1, SDP3XD2_I2C_ADDR },
+        { 0, SDP3XD2_I2C_ADDR },
+        { 2, SDP3XD2_I2C_ADDR },
+    };
+    bool found = false;
+    bool ret = false;
+    
+    for (uint8_t i=0; i<ARRAY_SIZE(addresses); i++)
+    {
+        _dev = hal.i2c_mgr->get_device(addresses[i].bus, addresses[i].addr);
+        if (!_dev)
+            continue;
+        if (!_dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER))
+            continue;
+
+        // lots of retries during probe
+        _dev->set_retries(10); 
+
+	// step 1 - reset on broadcast
+        uint8_t reset_cmd = SDP3X_RESET_CMD;
+        ret = _dev->transfer((uint8_t *)&reset_cmd, 1, nullptr, 0);
+        if (!ret)
+            printf("SDP3X: reset command failed\n");
+
+	// waiting 20ms to reset
+	hal.scheduler->delay_microseconds(20000);
+
+	// step 2 - configure
+        uint8_t cmd[2] = {(SDP3X_CONT_MEAS_AVG_MODE & 0xFF00) >> 8, SDP3X_CONT_MEAS_AVG_MODE & 0xFF};
+        ret = _dev->transfer((uint8_t *)&cmd, 2, nullptr, 0);
+        if (!ret)
+            printf("SDP3X: continued measure command failed\n");
+
+	// waiting again
+	hal.scheduler->delay_microseconds(20000);
+
+	// step 3 - get scale
+	uint8_t val[9];
+	ret = _dev->transfer(nullptr, 0, &val[0], sizeof(val));    
+        if (!ret)
+            printf("SDP3X: get scale value failed\n");
+
+    	// Check the CRC
+        if (!_crc(&val[0], 2, val[2]) || !_crc(&val[3], 2, val[5]) || !_crc(&val[6], 2, val[8]))
+	    printf("SDP3X: CRC failed !\n");
+
+	_scale = (((uint16_t)val[6]) << 8) | val[7];
+	printf("Scale value is : %d\n", _scale);
```

Comment:
Is this really needed? There is a message next that is different depending on this value.